File: backend/src/lakehouse/schemas/observability.py
from pydantic import BaseModel, Field


class PipelineStatus(BaseModel):
    status: str = Field(default="unknown")
    last_run: str = Field(default="")
    last_success: str = Field(default="")
    records_count: int = Field(default=0, ge=0)
    semaphore: str = Field(default="gray")


class PipelineLogs(BaseModel):
    lines: list[str] = Field(default_factory=list)
    total_lines: int = Field(default=0, ge=0)


class LayerRun(BaseModel):
    capa: str = Field(...)
    status: str = Field(default="unknown")
    run_id: str = Field(...)
    duracion_seg: float | None = Field(default=None)
    records_in: int = Field(default=0, ge=0)
    records_out: int = Field(default=0, ge=0)
    dlq_count: int = Field(default=0, ge=0)
    started_at: str | None = Field(default=None)
    finished_at: str | None = Field(default=None)


class PipelineLayersResponse(BaseModel):
    layers: list[LayerRun] = Field(default_factory=list)
    health_global: str = Field(default="sin datos")
    ultima_corrida_global: str = Field(default="")

File: frontend/src/api/observability.ts
import { apiClient } from './client'

export interface PipelineStatus {
  status: string
  last_run: string
  last_success: string
  records_count: number
  semaphore: string
}

export interface PipelineLogs {
  lines: string[]
  total_lines: number
}

export interface LayerRun {
  capa: string
  status: string
  run_id: string
  duracion_seg: number | null
  records_in: number
  records_out: number
  dlq_count: number
  started_at: string | null
  finished_at: string | null
}

export interface PipelineLayersResponse {
  layers: LayerRun[]
  health_global: string
  ultima_corrida_global: string
}

export async function getPipelineStatus(): Promise<PipelineStatus> {
  return apiClient<PipelineStatus>('/observability/status')
}

export async function getPipelineLogs(lines = 50): Promise<PipelineLogs> {
  return apiClient<PipelineLogs>(`/observability/logs?lines=${lines}`)
}

export async function getPipelineLayers(): Promise<PipelineLayersResponse> {
  return apiClient<PipelineLayersResponse>('/observability/pipeline/layers')
}

export async function getPipelineLogsByLayer(layer: string, lines = 50): Promise<PipelineLogs> {
  return apiClient<PipelineLogs>(`/observability/pipeline/logs/${layer}?lines=${lines}`)
}
